neuro_extras/main.py

```diff
@@ -1,119 +1,17 @@
+from typing import Any, MutableMapping
 
-import click
 import toml
-import yaml
-from neuromation import api as neuro_api
-from neuromation.api import ConfigError, Preset, Resources, find_project_root
-from neuromation.api.config import load_user_config
-from neuromation.api.parsing_utils import _as_repo_str
-from neuromation.api.url_utils import normalize_storage_path_uri, uri_from_cli
-from neuromation.cli.asyncio_utils import run as run_async
 from neuromation.cli.click_types import PresetType
-from neuromation.cli.const import EX_OK, EX_PLATFORMERROR
-from yarl import URL
 
-from neuro_extras.utils import get_neuro_client
-
-from .version import __version__
-
-
-NEURO_EXTRAS_IMAGE = os.environ.get(
-    "NEURO_EXTRAS_IMAGE", f"neuromation/neuro-extras:{__version__}"
-)
-
-SUPPORTED_ARCHIVE_TYPES = (
-    ".tar.gz",
-    ".tgz",
-    ".tar.bz2",
-    ".tbz2",
-    ".tbz",
-    ".tar",
-    ".gz",
-    ".zip",
-)
-
-SUPPORTED_OBJECT_STORAGE_SCHEMES = {
-    "AWS": "s3://",
-    "GCS": "gs://",
-    "AZURE": "azure+https://",
-    "HTTP": "http://",
-    "HTTPS": "https://",
-}
-
-
-@click.group()
-@click.version_option(
-    version=__version__, message="neuro-extras package version: %(version)s"
-)
-def main() -> None:
-    """
-    Auxiliary scripts and recipes for automating routine tasks.
-    """
-    handler = ClickLogHandler()
-    handler.setFormatter(logging.Formatter("%(levelname)s: %(message)s"))
-
-    root_logger = logging.getLogger()
-    root_logger.setLevel(logging.INFO)
-    root_logger.addHandler(handler)
-
-
-@main.group()
-def data() -> None:
-    """
-    Data transfer operations.
-    """
-    pass
-
-
-@data.command("transfer")
-@click.argument("source")
-@click.argument("destination")
-def data_transfer(source: str, destination: str) -> None:
-    """
-    Copy data between storages on different clusters.
-    """
-    run_async(_data_transfer(source, destination))
-
-
-@main.group()
-def image() -> None:
-    """
-    Job container image operations.
-    """
-    pass
-
-
-@image.command("transfer")
-@click.argument("source")
-@click.argument("destination")
-@click.option(
-    "-F",
-    "--force-overwrite",
-    default=False,
-    is_flag=True,
-    help="Transfer even if the destination image already exists.",
-)
-def image_transfer(source: str, destination: str, force_overwrite: bool) -> None:
-    """
-    Copy images between clusters.
-    """
-    exit_code = run_async(_image_transfer(source, destination, force_overwrite))
-    sys.exit(exit_code)
+from .cli import main
+from .config import config_save_docker_json  # noqa
+from .data import data_cp, data_transfer  # noqa
+from .image import image_build, image_transfer  # noqa
+from .k8s import generate_k8s_registry_secret, generate_k8s_secret  # noqa
+from .seldon import generate_seldon_deployment, seldon_init_package  # noqa
+from .upload_download import download, upload  # noqa
 
 
 @main.command("init-aliases")
```

Comment:
This command could be under the config group, but it might be the case, that we need to check dependencies (and probably, make it backward-compatible). WDYT?

Reply:
I'm pretty sure we've got quite a few scripts and tests relying upon this being a top-level command. Moving it to a config seems like a good idea, but not a kind of refactoring this PR introduces :)

Reply:
I've briefly checked our projects and found this command to be used only in tests and in the Dockerfile of neuro-extras, but not in neuro-cli or neuro-flow.
But you're right, it not a question for this PR, just wanted to know if my feeling regarding it is right :))